### PR TITLE
Enable ABSL_PROPAGATE_CXX_STD to ensure that C++ version requirements are propagated via CMake

### PR DESCRIPTION
**Pull request by traversaro**
This does not solve all possible ABI problems as this only put a constraint on the minimum version of the C++ standard version used in downstream projects, but it is still an improvement.

See discussion in https://github.com/conda-forge/mujoco-feedstock/pull/19#issuecomment-1431276538 .

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by traversaro** [comment]
@conda-forge-admin, please rerender

**Comment by h-vetinari** [comment]
Looks like we're running into an upstream bug. `ABSL_PROPAGATE_CXX_STD` got added in the last release AFAIR, so it's very possible that this doesn't work with windows (much less shared builds) yet.

**Comment by coryan** [comment]
It seems that `_NAME` is not defined, and we should use `_dll` here:

https://github.com/abseil/abseil-cpp/blob/lts_2023_01_25/CMake/AbseilDll.cmake#L790

and here:

https://github.com/abseil/abseil-cpp/blob/78be63686ba732b25052be15f8d6dee891c05749/CMake/AbseilDll.cmake#L800-L801

I will start the patches upstream, but I suspect they will not be backported.  Can we add a patch to the recipe?  I can help you with that, but may need some hints.

**Comment by traversaro** [comment]
Thanks @coryan ! A patch on top of all the other existing patches is https://github.com/traversaro/abseil-cpp/commit/9b7af55ec6cdd9126ba59c75fab1bea90d87da7d, I will add it to the PR.

**Comment by traversaro** [comment]
The fix suggested by @coryan is working fine, I thin the PR is ready for review @h-vetinari .

File: recipe/patches/0007-fix-export-cxx-std.patch [hunk 1]
```diff
@@ -0,0 +1,24 @@
+diff --git a/CMake/AbseilDll.cmake b/CMake/AbseilDll.cmake
```
Comment (h-vetinari) [inline recipe/patches/0007-fix-export-cxx-std.patch:1]: Could you apply this on top of the branch that I posted further up and then regenerate the patches with `git format-patch tags/20230125.0`?

Comment (traversaro) [reply recipe/patches/0007-fix-export-cxx-std.patch]: Done and opened PR at https://github.com/h-vetinari/abseil-cpp/pull/1 .

Comment (h-vetinari) [reply recipe/patches/0007-fix-export-cxx-std.patch]: Thanks a lot; though once it's in a format that `git am` understands, I don't need the PR actually, can just pick it up from the recipe 👍

**Comment by h-vetinari** [comment]
Thanks for the PR! :)

**Comment by h-vetinari** [comment]
> Thanks @coryan ! A patch on top of all the other existing patches is [traversaro/abseil-cpp@9b7af55](https://github.com/traversaro/abseil-cpp/commit/9b7af55ec6cdd9126ba59c75fab1bea90d87da7d), I will add it to the PR.

@traversaro, did you ever try to upstream that commit? I just saw an [issue](https://github.com/abseil/abseil-cpp/issues/1432) upstream that looks like it's running into the same problem. Would be good to try to upstream it (not least that we have less patches to carry, but also to fix it for everyone else of course) - I'd offer to do it for you, but you'll have to sign the google CLA, so I cannot offer your commit (as committer) in a PR.

**Comment by coryan** [comment]
AFAICT, it is already fixed upstream: https://github.com/abseil/abseil-cpp/commit/bd624d9f9825f76f14453beb3df81d82b9e17062

**Comment by h-vetinari** [comment]
> AFAICT, it is already fixed upstream: [abseil/abseil-cpp@bd624d9](https://github.com/abseil/abseil-cpp/commit/bd624d9f9825f76f14453beb3df81d82b9e17062)

Thanks for putting that in, hadn't seen that 👍

**Pull request merged**